### PR TITLE
Order all keys in config

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
Order all keys in config

## Motivation and Context
New changes

- return default values when sub-sys is
  not configured.
- the state is hidden parameter now
- remove worm mode to be saved in config

## How to test this PR?
you need an update `mc` as well 

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [x] Documentation needed
- [x] Unit tests needed
- [ ] Functional tests needed (If yes, add [mint](https://github.com/minio/mint) PR # here: )

### CODE REVIEW

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-xl.sh | :x: [more...](#2fbf2795-48ad-4277-9662-89b43b17e57a) |
| mint-large-bucket.sh | :x: [more...](#33e87621-58bc-455e-821e-289c2e594126) |
| mint-compression-fs.sh | :x: [more...](#e38cb4d7-a16f-4ebd-999f-2dec167218af) |
<a name="e38cb4d7-a16f-4ebd-999f-2dec167218af"></a>
### 8541-df6c06b/mint-compression-fs.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.51:30468
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp f389c0ef4f18:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 0 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 41 seconds
(4/14) Running aws-sdk-ruby tests ... done in 2 seconds
(5/14) Running awscli tests ... done in 54 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 38 seconds
(8/14) Running minio-dotnet tests ... done in 22 seconds
(9/14) Running minio-go tests ... done in 44 seconds
(10/14) Running minio-java tests ... done in 15 seconds
(11/14) Running minio-js tests ... done in 42 seconds
(12/14) Running minio-py tests ... done in 57 seconds
(13/14) Running s3cmd tests ... FAILED in 6 seconds
{
  "name": "s3cmd",
  "duration": "2495",
  "function": "test_get_object_multipart",
  "status": "FAIL",
  "error": "\n!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n    An unexpected error has occurred.\n  Please try reproducing the error using\n  the latest s3cmd code from the git master\n  branch found at:\n    https://github.com/s3tools/s3cmd\n  and have a look at the known issues list:\n    https://github.com/s3tools/s3cmd/wiki/Common-known-issues-and-their-solutions\n  If the error persists, please report the\n  following lines (removing any private\n  info as necessary) to:\n   s3tools-bugs@lists.sourceforge.net\n\n\n!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n\nInvoked as: /usr/local/bin/s3cmd --config /tmp/.s3cmd-8497/s3cfg get s3://s3cmd-test-bucket-18890/s3cmd-test-object-14804 s3cmd-test-object-14804.downloaded\nProblem: <class 'ConnectionResetError: [Errno 104] Connection reset by peer\nS3cmd:   2.0.2\npython:   3.5.2 (default, Oct  8 2019, 13:06:37) \n[GCC 5.4.0 20160609]\nenvironment LANG=C.UTF-8\n\nTraceback (most recent call last):\n  File \"/usr/local/bin/s3cmd\", line 3092, in <module>\n    rc = main()\n  File \"/usr/local/bin/s3cmd\", line 3001, in main\n    rc = cmd_func(args)\n  File \"/usr/local/bin/s3cmd\", line 590, in cmd_object_get\n    response = s3.object_get(uri, dst_stream, destination, start_position = start_position, extra_label = seq_label)\n  File \"/usr/local/lib/python3.5/dist-packages/S3/S3.py\", line 716, in object_get\n    response = self.recv_file(request, stream, labels, start_position)\n  File \"/usr/local/lib/python3.5/dist-packages/S3/S3.py\", line 1742, in recv_file\n    data = http_response.read(this_chunk)\n  File \"/usr/lib/python3.5/http/client.py\", line 458, in read\n    n = self.readinto(b)\n  File \"/usr/lib/python3.5/http/client.py\", line 498, in readinto\n    n = self.fp.readinto(b)\n  File \"/usr/lib/python3.5/socket.py\", line 575, in readinto\n    return self._sock.recv_into(b)\nConnectionResetError: [Errno 104] Connection reset by peer\n\n!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n    An unexpected error has occurred.\n  Please try reproducing the error using\n  the latest s3cmd code from the git master\n  branch found at:\n    https://github.com/s3tools/s3cmd\n  and have a look at the known issues list:\n    https://github.com/s3tools/s3cmd/wiki/Common-known-issues-and-their-solutions\n  If the error persists, please report the\n  above lines (removing any private\n  info as necessary) to:\n   s3tools-bugs@lists.sourceforge.net\n!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!"
}

Executed 12 out of 14 tests successfully.
```
<a name="33e87621-58bc-455e-821e-289c2e594126"></a>
### 8541-df6c06b/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.61:30902
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp ea758f63176e:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 5 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 53 seconds
(4/14) Running aws-sdk-ruby tests ... done in 5 seconds
(5/14) Running awscli tests ... done in 2 minutes and 22 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 1 minutes and 1 seconds
(8/14) Running minio-dotnet tests ... done in 1 minutes and 3 seconds
(9/14) Running minio-go tests ... FAILED in 3 minutes and 40 seconds
{
  "args": {},
  "duration": 91429,
  "error": "Please reduce your request",
  "function": "testStorageClassMetadataCopyObject()",
  "message": "CopyObject failed on RRS",
  "name": "minio-go: testStorageClassMetadataCopyObject",
  "status": "FAIL"
}

Executed 8 out of 14 tests successfully.
```
<a name="2fbf2795-48ad-4277-9662-89b43b17e57a"></a>
### 8541-df6c06b/mint-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.53:30697
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 0eaa8a9f84e6:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 6 minutes and 5 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-r9tqo2p3vy4dd9",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 364954,
  "error": "XMinioServerNotInitialized: Server not initialized, please try again.\n\tstatus code: 503, request id: 15D8BF7A0E583148, host id: ",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```
Error: No such image: minio/minio:8541-df6c06b

**Comment by poornas** [comment]
```
➜  minio git:(10b76751c) ./minio server ~/export                             
Server startup failed with 'Unable to initialize config system: json: cannot unmarshal object into Go value of type config.KVS'
Server switching to safe mode
Please use 'mc admin' commands to fix this issue
```
@harshavardhana, get above error with this PR. How do I get out of this? config.json was already encrypted and worked ok with commit 7cdb676.

**Comment by harshavardhana** [comment]
you need purge all your keys @poornas - the types have changed

**Comment by poornas** [comment]
there's an index out of range panic
```
mc admin config set myminio cache drives="/tmp/cache" quota=100
```
```
2019/11/19 19:31:21 http: panic serving 127.0.0.1:53498: runtime error: index out of range [0] with length 0
goroutine 736 [running]:
net/http.(*conn).serve.func1(0xc000498500)
	/usr/local/go/src/net/http/server.go:1767 +0x139
panic(0x195b980, 0xc00040a120)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/minio/minio/cmd.criticalErrorHandler.ServeHTTP.func1(0x1d67c60, 0xc00037a0e0, 0xc0001c8400)
	/home/kp/code/src/github.com/minio/minio/cmd/generic-handlers.go:765 +0x248
panic(0x195b980, 0xc00040a120)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/minio/minio/cmd/config.Config.SetKVS(0xc00051a180, 0xc0000b6390, 0x22, 0x1000, 0xc0000b6390)
	/home/kp/code/src/github.com/minio/minio/cmd/config/config.go:477 +0xf1a
github.com/minio/minio/cmd.adminAPIHandlers.SetConfigKVHandler(0x1d657e0, 0xc000498820, 0xc0001c8700)
	/home/kp/code/src/github.com/minio/minio/cmd/admin-handlers-config-kv.go:159 +0x60b
```

**Comment by harshavardhana** [comment]
Fixed @poornas

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_exclamation_mark: [more...](#f2e995ef-2466-4bcb-8a33-1b926a55417e) |
<a name="f2e995ef-2466-4bcb-8a33-1b926a55417e"></a>
### 8541-63a7160/mint-large-bucket.sh.xtrace:
```
DISABLED TIME BEING
```
Error: No such image: minio/minio:8541-63a7160

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-xl.sh | :x: [more...](#a97e19b1-4a5c-4585-a2e6-1da866f439aa) |
| mint-large-bucket.sh | :x: [more...](#fd81b249-968e-498b-9fda-43ab279f5add) |
<a name="fd81b249-968e-498b-9fda-43ab279f5add"></a>
### 8541-10b7675/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.51:32679
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 1ee98de2bfa2:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 1 seconds
(2/14) Running aws-sdk-java tests ... done in 2 seconds
(3/14) Running aws-sdk-php tests ... done in 55 seconds
(4/14) Running aws-sdk-ruby tests ... done in 8 seconds
(5/14) Running awscli tests ... done in 2 minutes and 15 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 1 minutes and 14 seconds
(8/14) Running minio-dotnet tests ... done in 47 seconds
(9/14) Running minio-go tests ... FAILED in 3 minutes and 13 seconds
{
  "args": {},
  "duration": 75527,
  "error": "Please reduce your request",
  "function": "testStorageClassMetadataCopyObject()",
  "message": "CopyObject failed on RRS",
  "name": "minio-go: testStorageClassMetadataCopyObject",
  "status": "FAIL"
}

Executed 8 out of 14 tests successfully.
```
<a name="a97e19b1-4a5c-4585-a2e6-1da866f439aa"></a>
### 8541-10b7675/mint-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.51:31836
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 0210e8fc7dbc:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 6 minutes and 4 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-rlgrs4e2wx2dsw",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 364652,
  "error": "XMinioServerNotInitialized: Server not initialized, please try again.\n\tstatus code: 503, request id: 15D8CC48994D7D25, host id: ",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```
Error: No such image: minio/minio:8541-10b7675

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :x: [more...](#02f9d2d8-9721-4d21-8eab-f92cf3a3202b) |
<a name="02f9d2d8-9721-4d21-8eab-f92cf3a3202b"></a>
### 8541-6e09f43/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.51:32461
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 8f8181d201ff:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 1 seconds
(2/14) Running aws-sdk-java tests ... done in 3 seconds
(3/14) Running aws-sdk-php tests ... done in 54 seconds
(4/14) Running aws-sdk-ruby tests ... done in 6 seconds
(5/14) Running awscli tests ... done in 1 minutes and 44 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 1 minutes and 31 seconds
(8/14) Running minio-dotnet tests ... done in 1 minutes and 4 seconds
(9/14) Running minio-go tests ... FAILED in 4 minutes and 1 seconds
{
  "args": {},
  "duration": 111076,
  "error": "Please reduce your request",
  "function": "testStorageClassMetadataCopyObject()",
  "message": "CopyObject failed on RRS",
  "name": "minio-go: testStorageClassMetadataCopyObject",
  "status": "FAIL"
}

Executed 8 out of 14 tests successfully.
```
Error: No such image: minio/minio:8541-6e09f43

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compression-xl.sh | :x: [more...](#ab3ddef7-825e-4641-8c90-d3f32cb30e32) |
| mint-large-bucket.sh | :x: [more...](#767ea85d-a7cf-47e1-933e-eafced4af0c2) |
<a name="ab3ddef7-825e-4641-8c90-d3f32cb30e32"></a>
### 8541-a7c48c7/mint-compression-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.53:31763
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp d939c272ee26:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 0 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 41 seconds
(4/14) Running aws-sdk-ruby tests ... done in 2 seconds
(5/14) Running awscli tests ... done in 1 minutes and 53 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 47 seconds
(8/14) Running minio-dotnet tests ... done in 28 seconds
(9/14) Running minio-go tests ... done in 35 seconds
(10/14) Running minio-java tests ... FAILED in 1 seconds
{
  "name": "minio-java",
  "function": "putObject(String bucketName, String objectName, String filename)",
  "args": "filename: 6MB",
  "duration": 641,
  "status": "FAIL",
  "error": "java.io.EOFException >>> [okio.RealBufferedSink.write(RealBufferedSink.java:115), io.minio.HttpRequestBody.writeTo(HttpRequestBody.java:77), okhttp3.internal.http.CallServerInterceptor.intercept(CallServerInterceptor.java:83), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147), okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.java:45), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121), okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.java:94), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121), okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.java:93), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147), okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.java:125), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121), okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:264), okhttp3.RealCall.execute(RealCall.java:93), io.minio.MinioClient.executeReq(MinioClient.java:1107), io.minio.MinioClient.execute(MinioClient.java:1066), io.minio.MinioClient.executePut(MinioClient.java:1430), io.minio.MinioClient.executePut(MinioClient.java:1452), io.minio.MinioClient.putObject(MinioClient.java:4511), io.minio.MinioClient.putObject(MinioClient.java:4608), io.minio.MinioClient.putObject(MinioClient.java:3879), FunctionalTest.putObject_test2(FunctionalTest.java:440), FunctionalTest.runTests(FunctionalTest.java:2985), FunctionalTest.main(FunctionalTest.java:3148)]"
}

Executed 9 out of 14 tests successfully.
```
<a name="767ea85d-a7cf-47e1-933e-eafced4af0c2"></a>
### 8541-a7c48c7/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.53:30009
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 7ea0711b827b:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 3 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 48 seconds
(4/14) Running aws-sdk-ruby tests ... done in 5 seconds
(5/14) Running awscli tests ... done in 2 minutes and 21 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 1 minutes and 17 seconds
(8/14) Running minio-dotnet tests ... done in 1 minutes and 11 seconds
(9/14) Running minio-go tests ... FAILED in 4 minutes and 20 seconds
{
  "args": {},
  "duration": 118579,
  "error": "Please reduce your request",
  "function": "testStorageClassMetadataCopyObject()",
  "message": "CopyObject failed on RRS",
  "name": "minio-go: testStorageClassMetadataCopyObject",
  "status": "FAIL"
}

Executed 8 out of 14 tests successfully.
```
Error: No such image: minio/minio:8541-a7c48c7

File: docs/disk-caching/DESIGN.md [hunk 1]
```diff
@@ -19,8 +19,8 @@ minio gateway <name> -h
   7. Start MinIO gateway to s3 with edge caching enabled on '/mnt/drive1', '/mnt/drive2' and '/mnt/export1 ... /mnt/export24',
      exclude all objects under 'mybucket', exclude all objects with '.pdf' as extension
      with expiry up to 40 days.
-     $ export MINIO_CACHE_DRIVES="/mnt/drive1;/mnt/drive2;/mnt/export{1..24}"
-     $ export MINIO_CACHE_EXCLUDE="mybucket/*;*.pdf"
+     $ export MINIO_CACHE_DRIVES="/mnt/drive1,/mnt/drive2,/mnt/export{1..24}"
```
Comment (kannappanr) [inline docs/disk-caching/DESIGN.md:22]: Is this a breaking change, the delimiter?

**Review by kannappanr — APPROVED** [review]
LGTM

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-xl.sh | :x: [more...](#e97d9d27-9c4c-44e4-8592-88c8bdbc5ced) |
| mint-large-bucket.sh | :x: [more...](#f8df86ba-7a13-4976-b3c6-2dc03caf2b40) |
| mint-compression-fs.sh | :x: [more...](#87301e27-311b-4b11-a06e-06c7b79b8b62) |
<a name="f8df86ba-7a13-4976-b3c6-2dc03caf2b40"></a>
### 8541-8dd26cb/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.61:32388
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp f1748bdb940f:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 1 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 46 seconds
(4/14) Running aws-sdk-ruby tests ... done in 4 seconds
(5/14) Running awscli tests ... done in 2 minutes and 7 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 1 minutes and 10 seconds
(8/14) Running minio-dotnet tests ... done in 57 seconds
(9/14) Running minio-go tests ... FAILED in 3 minutes and 55 seconds
{
  "args": {},
  "duration": 69901,
  "error": "Please reduce your request",
  "function": "testStorageClassMetadataCopyObject()",
  "message": "CopyObject failed on RRS",
  "name": "minio-go: testStorageClassMetadataCopyObject",
  "status": "FAIL"
}

Executed 8 out of 14 tests successfully.
```
<a name="e97d9d27-9c4c-44e4-8592-88c8bdbc5ced"></a>
### 8541-8dd26cb/mint-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.54:31255
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 04213a90f918:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 6 minutes and 6 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-vkyasclv5d0obj",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 365650,
  "error": "XMinioServerNotInitialized: Server not initialized, please try again.\n\tstatus code: 503, request id: 15D8FB5B3856F488, host id: ",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```
<a name="87301e27-311b-4b11-a06e-06c7b79b8b62"></a>
### 8541-8dd26cb/mint-compression-fs.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.59:32270
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 12210edee566:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 0 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 41 seconds
(4/14) Running aws-sdk-ruby tests ... done in 3 seconds
(5/14) Running awscli tests ... done in 1 minutes and 50 seconds
(6/14) Running healthcheck tests ... done in 1 seconds
(7/14) Running mc tests ... done in 35 seconds
(8/14) Running minio-dotnet tests ... done in 22 seconds
(9/14) Running minio-go tests ... done in 40 seconds
(10/14) Running minio-java tests ... done in 14 seconds
(11/14) Running minio-js tests ... FAILED in 14 seconds
{
  "name": "minio-js",
  "function": "fGetObject(bucketName, objectName, filePath)",
  "args": "bucketName:minio-js-test-3d5034ad-e3ad-4cf4-9a78-ef65114d5f9a, objectName:datafile-65-MB, filePath:/tmp/datafile-65-MB.download",
  "duration": 279,
  "status": "FAIL",
  "error": "Error: Size mismatch between downloaded file and the object at /mint/run/core/minio-js/node_modules/minio/dist/main/minio.js:965:12 at nextTask (node_modules/minio/node_modules/async/dist/async.js:4578:13) at next (node_modules/minio/node_modules/async/dist/async.js:4586:13) at /mint/run/core/minio-js/node_modules/minio/node_modules/async/dist/async.js:325:20 at FSReqWrap.oncomplete (fs.js:123:15)"
}

Executed 10 out of 14 tests successfully.
```
Error: No such image: minio/minio:8541-8dd26cb

**Review by poornas — APPROVED** [review]

**Pull request merged**